src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -62,6 +64,22 @@ export class DatasetService {
     return this.defaultDataset;
   }
 
+  public getRecentDataset(dataset: string | undefined): Dataset | undefined {
+    if (dataset) {
+      return this.recentDatasets.get(dataset);
+    }
+  }
+
+  public getRecentDatasets(): Dataset[] {
```

Comment:
it would be nice if we can specify a number a limit to that, default can be no limit. This will help us show this in other places such as the advanced selector without modifications

Reply:
+1, UI settings is already passed in. we could just add a new advanced setting